src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -0,0 +1,605 @@
+        sage: p1.tropical_variety()
+        Tropical curve of 3*a*b + 0*a + (-1)*b
+        sage: p1.tropical_variety().plot()
+        Graphics object consisting of 3 graphics primitives
+
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=True)
+        R = PolynomialRing(T, ('a,b'))
+        a, b = R.gen(), R.gen(1)
+        p1 = R(3)*a*b + a + R(-1)*b
+        tv1 = p1.tropical_variety()
+        sphinx_plot(tv1.plot())
+
+    Tropical polynomial in two variables will induce a function in three
+    dimension that consists of a number of surfaces::
+
+        sage: p1.plot3d()
+        Graphics3d Object
+
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=True)
+        R = PolynomialRing(T, ('a,b'))
+        a, b = R.gen(), R.gen(1)
+        p1 = R(3)*a*b + a + R(-1)*b
+        sphinx_plot(p1.plot3d())
+
+    If we use a max-plus algebra, we will get a slightly different result::
+
+        sage: T = TropicalSemiring(QQ, use_min=False)
+        sage: R.<a,b> = PolynomialRing(T)
+        sage: p1 = R(3)*a*b + a + R(-1)*b
+        sage: p1.tropical_variety()
+        Tropical curve of 3*a*b + 0*a + (-1)*b
+        sage: p1.tropical_variety().plot()
+        Graphics object consisting of 3 graphics primitives
+
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=False)
+        R = PolynomialRing(T, ('a,b'))
+        a, b = R.gen(), R.gen(1)
+        p1 = R(3)*a*b + a + R(-1)*b
+        tv1 = p1.tropical_variety()
+        sphinx_plot(tv1.plot())
+
+    ::
+
+        sage: p1.plot3d()
+        Graphics3d Object
+
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=False)
+        R = PolynomialRing(T, ('a,b'))
+        a, b = R.gen(), R.gen(1)
+        p1 = R(3)*a*b + a + R(-1)*b
+        sphinx_plot(p1.plot3d())
+
+    TESTS:
+
+    There is no subtraction defined for tropical polynomials::
+
+        sage: T = TropicalSemiring(QQ)
+        sage: R.<a,b> = PolynomialRing(T)
+        sage: a - b
+        Traceback (most recent call last):
+        ...
+        ArithmeticError: cannot negate any non-infinite element
+    """
+    def subs(self, fixed=None, **kwds):
+        r"""
+        Fix some given variables in ``self`` and return the changed
+        tropical multivariate polynomials.
+
+        OUTPUT: new :class:`TropicalMPolynomial`
+
+        .. SEEALSO::
+
+            :meth:`sage.rings.polynomial.multi_polynomial_element.MPolynomial_polydict.subs`
+
+        EXAMPLES::
+
+            sage: T = TropicalSemiring(QQ, use_min=False)
+            sage: R.<x,y> = PolynomialRing(T)
+            sage: p1 = x^2 + y + R(3)
+            sage: p1((R(4),y))
+            0*y + 8
+            sage: p1.subs({x: 4})
+            0*y + 8
+        """
+        def check_type(dict, var):
+            try:
+                variables[i] = T(dict[var])
+            except TypeError:
+                variables[i] = dict[var]
```

Comment:
You should not be converting the input to tropical values. The user should know what they want, and they might want to pass something that is convertible to a tropical ring element but they do not want it to be treated as such (e.g., a constant tropical polynomial). Make the user be explicit with things is better than any potential convenience.